Automattic/jetpack
# E2E tests: split tunnel setup

#### Changes proposed in this Pull Request:

Split the tunnel management from the test execution. Until now, the tunnel was created and closed within the tests (global setup and global teardown). 

For more flexibility, this PR extracts the tunnel management in a script that can be used standalone, or called from within the tests as before, if needed.

The tunnel is managed by `bin/tunnel.js` script.
[PM2](https://pm2.keymetrics.io/) is used to manage the tunnel process and keep it running in the background, so we can get back the CLI to execute the tests in CI.

This is a basic working implementation that can be further expanded if needed. 

#### Testing instructions:
- CI happy
- Locally:
  - Start a tunnel with `yarn tunnel-on`
    - If `config/tmp/e2e-tunnels.txt` exists and contains a valid URL a tunnel is created with the same subdomain.
    - If `config/tmp/e2e-tunnels.txt` doesn't exist a tunnel is created with new subdomain and url is written in file.
    - If tunnel is already running the process is restarted, the result should be an active tunnel with the same URL.
  - Check running tunnel process with `yarn pm2 list`.
  - Stop the tunnel with `yarn tunnel-off`.

#### Jetpack product discussion

1156386383419466-as-1200136224519585

#### Does this pull request change what data or activity we track or use?
no
